src/utils.ts

```diff
@@ -36,10 +36,34 @@ function timestampToNanos(timestamp: bigint, unit: TimestampUnit) {
   }
 }
 
+/**
+ * Fetches JSON data from a URL.
+ * @template T - The expected type of the JSON response
+ * @param url - The URL to fetch from
+ * @returns Promise resolving to the parsed JSON data
+ * @throws Error if the request fails or returns a non-OK status
+ */
+async function fetchJson<T>(url: string): Promise<T> {
+  let response: globalThis.Response;
+  try {
+    response = await fetch(url);
```

Comment:
nit: we could use the same http lib/agent as what's defined in the config.

Reply:
node 18+ uses undici fetch() by default.
unlikely that anyone would need to switch this to the fetch() from the http/https modules.